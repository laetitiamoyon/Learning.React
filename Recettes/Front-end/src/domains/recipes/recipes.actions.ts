import { Ingredient } from '../ingredients/ingredients.model';
import { Recipe, RemoveRecipeIngredientSuccessPayload, UpdateRecipeIngredientSuccessPayload } from './recipes.model';

export enum RecipeAction
{
    GET_RECIPES_REQUEST = "RECIPES/GET_RECIPE_REQUEST",
    GET_RECIPES_SUCCESS = "RECIPES/GET_RECIPE_SUCCESS",
    GET_RECIPES_FAILED = "RECIPES/GET_RECIPE_FAILED",

    ADD_RECIPE_REQUEST = "RECIPES/ADD_RECIPE_REQUEST",
    ADD_RECIPE_SUCCESS = "RECIPES/ADD_RECIPE_SUCCESS",
    ADD_RECIPE_FAILED = "RECIPES/ADD_RECIPE_FAILED",

    EDIT_RECIPE_REQUEST = "RECIPES/EDIT_RECIPE_REQUEST",
    EDIT_RECIPE_SUCCESS = "RECIPES/EDIT_RECIPE_SUCCESS",
    EDIT_RECIPE_FAILED = "RECIPES/EDIT_RECIPE_FAILED",

    UPDATE_RECIPE_INGREDIENT_REQUEST = "RECIPES/UPDATE_RECIPE_INGREDIENT_REQUEST",
    UPDATE_RECIPE_INGREDIENT_SUCCESS = "RECIPES/UPDATE_RECIPE_INGREDIENT_SUCCESS",
    UPDATE_RECIPE_INGREDIENT_FAILED = "RECIPES/UPDATE_RECIPE_INGREDIENT_FAILED",

    REMOVE_RECIPE_REQUEST = "RECIPES/REMOVE_RECIPE_REQUEST",
    REMOVE_RECIPE_SUCCESS = "RECIPES/REMOVE_RECIPE_SUCCESS",
    REMOVE_RECIPE_FAILED = "RECIPES/REMOVE_RECIPE_FAILED",

    REMOVE_RECIPE_INGREDIENT_REQUEST = "RECIPES/REMOVE_RECIPE_INGREDIENT_REQUEST",
    REMOVE_RECIPE_INGREDIENT_SUCCESS = "RECIPES/REMOVE_RECIPE_INGREDIENT_SUCCESS",
    REMOVE_RECIPE_INGREDIENT_FAILED = "RECIPES/REMOVE_RECIPE_INGREDIENT_FAILED",

    SET_INFORMATION_MESSAGE = "SET_INFORMATION_MESSAGE"
}

export interface GetRecipesRequestAction
{ 
    type: RecipeAction.GET_RECIPES_REQUEST
}

export interface GetRecipesSuccessAction
{ 
    type: RecipeAction.GET_RECIPES_SUCCESS
    payload : Recipe[]
}

export interface GetRecipesFailedAction
{ 
    type: RecipeAction.GET_RECIPES_FAILED
    error : string
}

export interface AddRecipeRequestAction
{ 
    type: RecipeAction.ADD_RECIPE_REQUEST
    payload : Recipe
}

export interface AddRecipeSuccessAction
{ 
    type: RecipeAction.ADD_RECIPE_SUCCESS
    payload : Recipe
}

export interface AddRecipeFailedAction
{ 
    type: RecipeAction.ADD_RECIPE_FAILED
    error : string
}

export interface EditRecipeRequestAction
{ 
    type: RecipeAction.EDIT_RECIPE_REQUEST
    payload : Recipe
}

export interface EditRecipeSuccessAction
{ 
    type: RecipeAction.EDIT_RECIPE_SUCCESS
    payload : Recipe
}

export interface EditRecipeFailedAction
{ 
    type: RecipeAction.EDIT_RECIPE_FAILED
    error : string
}

export interface UpdateRecipeIngredientRequestAction
{ 
    type: RecipeAction.UPDATE_RECIPE_INGREDIENT_REQUEST
    payload : Ingredient
}

export interface UpdateRecipeIngredientSuccessAction
{ 
    type: RecipeAction.UPDATE_RECIPE_INGREDIENT_SUCCESS
    payload : UpdateRecipeIngredientSuccessPayload
}

export interface UpdateRecipeIngredientFailedAction
{ 
    type: RecipeAction.UPDATE_RECIPE_INGREDIENT_FAILED
    error : string
}

export interface RemoveRecipeRequestAction
{ 
    type: RecipeAction.REMOVE_RECIPE_REQUEST
    payload : { id : string }
}

export interface RemoveRecipeSuccessAction
{ 
    type: RecipeAction.REMOVE_RECIPE_SUCCESS
    payload : { id : string }
}

export interface RemoveRecipeFailedAction
{ 
    type: RecipeAction.REMOVE_RECIPE_FAILED
    error : string
}

export interface RemoveRecipeIngredientRequestAction
{ 
    type: RecipeAction.REMOVE_RECIPE_INGREDIENT_REQUEST
    payload : { ingredientId : string }
}

export interface RemoveRecipeIngredientSuccessAction
{ 
    type: RecipeAction.REMOVE_RECIPE_INGREDIENT_SUCCESS
    payload : RemoveRecipeIngredientSuccessPayload
}

export interface RemoveRecipeIngredientFailedAction
{ 
    type: RecipeAction.REMOVE_RECIPE_INGREDIENT_FAILED
    error : string
}

export interface SetInformationMessageAction
{
    type : RecipeAction.SET_INFORMATION_MESSAGE,
    payload? : string
}

export const getRecipesRequestAction = () : GetRecipesRequestAction => ({
    type : RecipeAction.GET_RECIPES_REQUEST,
})

export const getRecipesSuccessAction = (payload : Recipe[]) : GetRecipesSuccessAction => ({
    type : RecipeAction.GET_RECIPES_SUCCESS,
    payload
})

export const getRecipesFailedAction = (error : string) : GetRecipesFailedAction => ({
    type : RecipeAction.GET_RECIPES_FAILED,
    error
})

export const addRecipeRequestAction = (payload : Recipe) : AddRecipeRequestAction => ({
    type : RecipeAction.ADD_RECIPE_REQUEST,
    payload
})

export const addRecipeSuccessAction = (payload : Recipe) : AddRecipeSuccessAction => ({
    type : RecipeAction.ADD_RECIPE_SUCCESS,
    payload
})

export const addRecipeFailedAction = (error : string) : AddRecipeFailedAction => ({
    type : RecipeAction.ADD_RECIPE_FAILED,
    error
})

export const editRecipeRequestAction = (recipe : Recipe) : EditRecipeRequestAction => ({
    type : RecipeAction.EDIT_RECIPE_REQUEST,
    payload : recipe
})

export const editRecipeSuccessAction = (payload : Recipe) : EditRecipeSuccessAction => ({
    type : RecipeAction.EDIT_RECIPE_SUCCESS,
    payload 
})

export const editRecipeFailedAction = (error : string) : EditRecipeFailedAction => ({
    type : RecipeAction.EDIT_RECIPE_FAILED,
    error
})

export const updateRecipeIngredientRequestAction = (ingredient : Ingredient) : UpdateRecipeIngredientRequestAction => ({
    type : RecipeAction.UPDATE_RECIPE_INGREDIENT_REQUEST,
    payload : ingredient
})

export const updateRecipeIngredientSuccessAction = (payload : UpdateRecipeIngredientSuccessPayload) : UpdateRecipeIngredientSuccessAction => ({
    type : RecipeAction.UPDATE_RECIPE_INGREDIENT_SUCCESS,
    payload 
})

export const updateRecipeIngredientFailedAction = (error : string) : UpdateRecipeIngredientFailedAction => ({
    type : RecipeAction.UPDATE_RECIPE_INGREDIENT_FAILED,
    error
})

export const removeRecipeRequestAction = (id: string) : RemoveRecipeRequestAction => ({ 
    type: RecipeAction.REMOVE_RECIPE_REQUEST,
    payload : { id } 
})

export const RemoveRecipeSuccessAction = (id: string) : RemoveRecipeSuccessAction => ({ 
    type: RecipeAction.REMOVE_RECIPE_SUCCESS,
    payload : { id } 
})

export const removeRecipeFailedAction = (error : string) : RemoveRecipeFailedAction => ({ 
    type: RecipeAction.REMOVE_RECIPE_FAILED,
    error 
})

export const removeRecipeIngredientRequestAction = (ingredientId: string) : RemoveRecipeIngredientRequestAction => ({ 
    type: RecipeAction.REMOVE_RECIPE_INGREDIENT_REQUEST,
    payload : { ingredientId } 
})

export const removeRecipeIngredientSuccessAction = (payload : RemoveRecipeIngredientSuccessPayload) : RemoveRecipeIngredientSuccessAction => ({ 
    type: RecipeAction.REMOVE_RECIPE_INGREDIENT_SUCCESS,
    payload
})

export const removeRecipeIngredientFailedAction = (error : string) : RemoveRecipeIngredientFailedAction => ({ 
    type: RecipeAction.REMOVE_RECIPE_INGREDIENT_FAILED,
    error 
})

export const setInformationMessageAction = (informationMessage? : string) : SetInformationMessageAction => ({
    type : RecipeAction.SET_INFORMATION_MESSAGE,
    payload : informationMessage
})

export type RecipeActions = 
    GetRecipesRequestAction |
    GetRecipesSuccessAction |
    GetRecipesFailedAction |

    AddRecipeRequestAction |
    AddRecipeSuccessAction |
    AddRecipeFailedAction |
    
    EditRecipeRequestAction |
    EditRecipeSuccessAction |
    EditRecipeFailedAction |

    UpdateRecipeIngredientRequestAction |
    UpdateRecipeIngredientSuccessAction |
    UpdateRecipeIngredientFailedAction |
    
    RemoveRecipeRequestAction | 
    RemoveRecipeSuccessAction | 
    RemoveRecipeFailedAction |

    RemoveRecipeIngredientRequestAction | 
    RemoveRecipeIngredientSuccessAction | 
    RemoveRecipeIngredientFailedAction |

    SetInformationMessageAction