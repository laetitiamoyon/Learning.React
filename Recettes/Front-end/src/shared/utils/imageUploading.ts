import { ChangeEvent } from 'react';

export const imageUploading = (
  event : ChangeEvent<HTMLInputElement>,
  onImageUploaded : (image: string) => void
) =>
{
  if (!event.target.files || !event.target.files[0]) return

  let file = event.target.files[0]
  let reader = new FileReader()

  reader.onload = (event : ProgressEvent<FileReader>) => onImageUploaded(event?.target?.result as string)
  reader.readAsDataURL(file)
};
