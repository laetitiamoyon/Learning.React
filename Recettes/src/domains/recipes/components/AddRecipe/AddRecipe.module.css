.addRecipeContainer
{
    background-image: url(./images/background-addRecipe.jpg);
    background-size: cover;
    
    margin: auto;
    padding: 0 24px;

    height: 100vh;

}

.formTitle
{ 
    color: white;
    text-align: center;
    font-size: 28px;
    text-align: center;

    padding-top: calc(8vh + 32px);
    margin-bottom: 32px;
    
    text-shadow: 2px 2px 4px black;
}

.formContainer { display: flex; flex-direction: column; }

.formElementContainer { display: flex; justify-content: center; }

.submitButton
{
    width: 130px;
    height: 30px;

    color: #394044;
    border: #BDBDBD solid 1px;
    border-radius: 8px;

    font-weight: 600;

    margin-top: 32px;
}
.submitButton:hover {
    background-color: #D15E3F;
    border:  #D15E3F;
    color: white;
}


.addIngredientButton 
{
    width: 50px;
    height: 25px;

    line-height: 20px;
    font-size: 18px;


    color: #394044;
    border: #BDBDBD solid 1px;
    border-radius: 8px;

    font-weight: 600;
}

.input
{
    width: 500px;
    color: #394044;
    border: #BDBDBD solid 1px;
    border-radius: 8px;
    padding: 6px;
}

.title { margin-top: 20px;}