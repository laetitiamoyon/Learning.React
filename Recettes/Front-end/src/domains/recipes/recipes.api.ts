import axios, { AxiosResponse, AxiosRequestConfig } from 'axios'
import { Recipe } from './recipes.model'

export const url : string = 'http://localhost:3001/recipes'
export const configuration : AxiosRequestConfig = { headers : { 'Content-Type' : 'application/json' } }


export const getAllRecipes = async () : Promise<AxiosResponse<Recipe[]>> =>
    (await axios.get(url, configuration)).data

export const addRecipe = async (recipe : Recipe) : Promise<AxiosResponse<Recipe>> =>
    (await axios.post(url, recipe, configuration)).data

export const updateRecipe = async (recipe : Recipe) : Promise<AxiosResponse<Recipe>> => 
    (await axios.put(`${url}/${recipe.id}`, recipe, configuration)).data

export const removeRecipe = async (id : string) : Promise<AxiosResponse<Recipe>> =>
    (await axios.delete(`${url}/${id}`, configuration)).data