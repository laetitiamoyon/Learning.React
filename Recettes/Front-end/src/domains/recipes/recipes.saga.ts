import { call, put, takeLatest } from 'redux-saga/effects'
import { getRecipesFailedAction, AddRecipeRequestAction, addRecipeSuccessAction, addRecipeFailedAction, EditRecipeRequestAction, RemoveRecipeRequestAction, RemoveRecipeSuccessAction, removeRecipeFailedAction, getRecipesSuccessAction, editRecipeSuccessAction, editRecipeFailedAction, RecipeAction, getRecipesRequestAction, UpdateRecipeIngredientRequestAction, updateRecipeIngredientSuccessAction, updateRecipeIngredientFailedAction, removeRecipeIngredientSuccessAction, removeRecipeIngredientFailedAction, RemoveRecipeIngredientRequestAction, setInformationMessageAction } from './recipes.actions';
import { Recipe } from './recipes.model';
import { addRecipe, getAllRecipes, updateRecipe, removeRecipe } from './recipes.api';

const computeInformationMessage = (recipes : Recipe[]) : string | undefined => 
{
   const recipesWithoutIngredient = recipes.filter(r => r.ingredients.length === 0)

   return recipesWithoutIngredient.length === 0 ? undefined :
   "les recettes suivantes ont été supprimées :" +
   recipesWithoutIngredient.map(r => r.title).join(',')
}

export function* getAllRecipesSaga()
{
   try
   {
      const recipes : Recipe[] = yield call(getAllRecipes);
     
      yield put(getRecipesSuccessAction(recipes));
   }
   catch (error : any)
   {
      yield put(getRecipesFailedAction(error.message));
   }
}

export function* addRecipeSaga(action : AddRecipeRequestAction)
{
   try
   {
      const newRecipe : Recipe = yield call(addRecipe, action.payload);
    
      yield put(addRecipeSuccessAction(newRecipe));
      yield put(getRecipesRequestAction());
   }
   catch (error : any)
   {
      yield put(addRecipeFailedAction(error.message));
   }
}

export function* editRecipeSaga(action : EditRecipeRequestAction)
{
   try
   {      
      const editedRecipe : Recipe = yield call(updateRecipe, action.payload);
      
      yield put(editRecipeSuccessAction(editedRecipe));
      yield put(getRecipesRequestAction());
   }
   catch (error : any)
   {
      yield put(editRecipeFailedAction(error.message));
   }
}

export function* updateRecipeIngredientSaga(action : UpdateRecipeIngredientRequestAction)
{
   try
   {
      const recipes : Recipe[] = yield call(getAllRecipes);
      const ingredient = action.payload

      // Récupérer les recettes qui contiennent l'ingrédient à modifier
      const recipesToUpdated = recipes.filter(r => r.ingredients.find(i => i.id === ingredient.id))

      // On met à jour l'ingrédient des recettes qui le contienne
      const updatedRecipes = recipesToUpdated.map(r => 
      ({
         ...r,
         ingredients : r.ingredients.map(i => ingredient.id === i.id ? ingredient : i)
      }))
      
      // On demande à l'API de mettre à jour les données 
      for (let updatedRecipe of updatedRecipes)
      {
         if (updatedRecipe.ingredients.length === 0) 
            yield call(removeRecipe, updatedRecipe.id);
         else 
            yield call(updateRecipe, updatedRecipe);
         
            yield put(updateRecipeIngredientSuccessAction({
            recipeId : updatedRecipe.id,
            ingredient
         }))
      }

      const informationMessage = computeInformationMessage(updatedRecipes)
      yield put (setInformationMessageAction(informationMessage))
      yield put(getRecipesRequestAction());
   }
   catch (error : any)
   {
      yield put(updateRecipeIngredientFailedAction(error.message));
   }
}

export function* removeRecipeSaga(action : RemoveRecipeRequestAction)
{
   try
   {
      const { id } = action.payload

      yield call(removeRecipe, id);
      yield put(RemoveRecipeSuccessAction(id));
      yield put(getRecipesRequestAction());
   }

   catch (error : any)
   {
      yield put(removeRecipeFailedAction(error.message));
   }
}

export function* removeRecipeIngredientSaga(action : RemoveRecipeIngredientRequestAction)
{
   try
   {
      const recipes : Recipe[] = yield call(getAllRecipes);
      const {ingredientId} = action.payload

      // Récupérer les recettes contenant l'ingrédient 
      const recipesToUpdated = recipes.filter(r => r.ingredients.find(i => i.id === ingredientId))

      // Supprimer l'ingrédient présent dans les recettes 
      const updatedRecipes = recipesToUpdated.map(r => ({
         ...r,
         ingredients: r.ingredients.filter(i => ingredientId !== i.id)
      }))

      for (let updatedRecipe of updatedRecipes)
      {
         if (updatedRecipe.ingredients.length === 0)
            yield call(removeRecipe, updatedRecipe.id);
         else
            yield call(updateRecipe, updatedRecipe);

         yield put (removeRecipeIngredientSuccessAction({
            recipeId : updatedRecipe.id,
            ingredientId
         }))

      }

      const informationMessage = computeInformationMessage(updatedRecipes)
      yield put (setInformationMessageAction(informationMessage))
      yield put(getRecipesRequestAction());
   }
   catch (error : any)
   {
      yield put(removeRecipeIngredientFailedAction(error.message));
   }
}

export default function* recipeSaga() : Generator {
  yield takeLatest(RecipeAction.GET_RECIPES_REQUEST, getAllRecipesSaga);
  yield takeLatest(RecipeAction.ADD_RECIPE_REQUEST, addRecipeSaga);
  yield takeLatest(RecipeAction.EDIT_RECIPE_REQUEST, editRecipeSaga);
  yield takeLatest(RecipeAction.UPDATE_RECIPE_INGREDIENT_REQUEST, updateRecipeIngredientSaga);
  yield takeLatest(RecipeAction.REMOVE_RECIPE_REQUEST, removeRecipeSaga);
  yield takeLatest(RecipeAction.REMOVE_RECIPE_INGREDIENT_REQUEST, removeRecipeIngredientSaga);
}
