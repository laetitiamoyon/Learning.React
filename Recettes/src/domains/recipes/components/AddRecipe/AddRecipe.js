import React, { useState, useContext} from 'react';
import styles from './AddRecipe.module.css'
import { useHistory } from 'react-router-dom';
import { RecipeContext } from '../../recipes.provider';

const NewRecipe = () => 
{
    const [title, setTitle] = useState('')
    const [description, setDescription] = useState('')
    const { dispatch } = useContext(RecipeContext)
    const history = useHistory();

    const redirectToRecipes = () => history.push('/recettes')
    const addRecipe = () => dispatch({ type: 'ADD_RECIPE', payload : { title, description } })

    const onTitleChange = (event) => setTitle(event.target.value)
    const onDescriptionChange = (event) => setDescription(event.target.value)

    const onSubmit = (event) =>
    {
        event.preventDefault()

        addRecipe()
        redirectToRecipes();
    }

    return <>
        <h1 className={styles.formTitle}>Créer une nouvelle recette</h1>
        <form className={styles.formContainer} onSubmit={onSubmit}>
            <label className={styles.title}>Titre:</label>
            <input 
                onChange={onTitleChange}
                className={styles.recipeTitle}
                name="titre"
                value={title}/>
            
            <label className={styles.title}> Description:</label>
            <textarea 
                onChange={onDescriptionChange}
                rows="5"
                className={styles.recipeDescription}
                name="Description"
                value={description}/>

            <label className={styles.title}>Ingrédients </label>
            <select className={styles.selectIngredient} name="ingredients"></select>

            <button className={styles.submitButton}>Enregistrer</button>
        </form>
    </>
}

export default NewRecipe;