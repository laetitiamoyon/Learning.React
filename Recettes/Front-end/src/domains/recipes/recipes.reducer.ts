import { failedActionMetadata, loadedActionMetadata, loadingActionMetadata } from '../../shared/domains/Redux/redux.utils';
import { RecipeAction, RecipeActions } from './recipes.actions';
import { initialRecipes, RecipeState } from './recipes.state';

export const recipesReducer = (state: RecipeState = initialRecipes , action: RecipeActions): RecipeState =>
{
    switch (action.type)
    {
        case RecipeAction.GET_RECIPES_REQUEST : return loadingActionMetadata(state)
        case RecipeAction.GET_RECIPES_SUCCESS : return {
            ...loadedActionMetadata(state),
            recipes: action.payload
        }
        case RecipeAction.GET_RECIPES_FAILED : return failedActionMetadata(action.error, state)

        case RecipeAction.SET_INFORMATION_MESSAGE : return { ...state, informationMessage : action.payload }

        default : return state
    }
}