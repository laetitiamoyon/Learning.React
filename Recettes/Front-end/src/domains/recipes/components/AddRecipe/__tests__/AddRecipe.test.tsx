import {fireEvent, render} from "@testing-library/react";
import {useDispatch, useSelector} from "react-redux";
import React from "react";
import AddRecipe from "../AddRecipe";
import {addRecipeRequestAction} from "../../../recipes.actions";
import {newGuid} from "../../../../../shared/utils/string";
import {useHistory} from "react-router-dom";
import {recipeStateMock} from "../../../recipes.mock";
import {ingredientStateMock} from "../../../../ingredients/ingredient.mock";

jest.mock('../../../../../shared/utils/string')
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({ push: jest.fn()})
}));
jest.mock('react-redux')
const mockUseDispatch = useDispatch as jest.MockedFunction<typeof useDispatch>
const mockDispatch = jest.fn()

const history = useHistory()
const mockRedirectToRecipeList = jest.fn(history.push)


describe("AddRecipe", () => {

    beforeEach(() => (mockUseDispatch as jest.Mock).mockReturnValue(mockDispatch))

    it("should match snapshot", async () => {
        // Given
        (useSelector as jest.Mock).mockReturnValue(recipeStateMock)

        // When
        const {container} = render(<AddRecipe/>)


        // Then
        expect(container).toMatchSnapshot()
    })

    it("should dispatch addIngredientRequestAction when submit button is clicked", async () =>
    {
        // Given
        (useSelector as jest.Mock).mockReturnValue(ingredientStateMock)
        const { container } = render(<AddRecipe />)
        const button = container.getElementsByTagName('button')[0]

        // When
        const consistentGuid = newGuid();
        (newGuid as jest.Mock).mockReturnValue(consistentGuid)
        fireEvent.click(button)

        // Then
        expect(mockDispatch).toBeCalledWith(addRecipeRequestAction({
            id : consistentGuid,
            title : '',
            description : '',
            ingredients : [],
            preparationTime: "",
            image: "",
            calories: "",
            cookingTime: ""

        }));
        expect(mockRedirectToRecipeList).toHaveBeenCalled()
        expect(mockUseDispatch).toHaveBeenCalled()
    })

})