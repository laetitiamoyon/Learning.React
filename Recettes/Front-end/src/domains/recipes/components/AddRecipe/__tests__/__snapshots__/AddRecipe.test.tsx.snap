// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe should match snapshot 1`] = `
<div>
  <div
    class="containerPage"
  >
    <h1
      class="title"
    >
      Ajouter une nouvelle recette
    </h1>
    <form
      class="form"
    >
      <div
        class="container"
      >
        <div
          class="container"
        >
          <h1
            class="title"
          >
            Choisissez l'image de votre recette
          </h1>
          <label
            class="label"
            for="imageUpload"
          >
            Télécharger l'image
          </label>
          <input
            accept="image/png, image/jpeg"
            class="input"
            id="imageUpload"
            type="file"
          />
          <img
            alt=""
            class="image"
            src=""
          />
        </div>
        <div
          class="element"
        >
          <div
            class="rightElements"
          >
            <label
              class="label"
            >
              Titre de la recette
            </label>
            <input
              class="input"
              name="Titre de la recette"
              type="text"
              value=""
            />
            <label
              class="label"
            >
              Description
            </label>
            <textarea
              class="input"
              name="Description"
              rows="5"
            />
            <div
              class="addRecipeIngredient"
            >
               
              <div
                class="container"
              >
                <button
                  class="addIngredientButton"
                >
                  Ajouter un ingrédient
                </button>
                <div
                  class="selectAnIngredient"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <h1
          class="title"
        >
          Autres informations
        </h1>
        <div
          class="elements"
        >
          <svg
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32.202 30.743c.453-.344.461-1.004.057-1.404L10.234 7.55a1.904 1.904 0 0 0-2.679 0 1.864 1.864 0 0 0 0 2.65l5.803 5.74-4.463 4.418 9.82 9.716a10.155 10.155 0 0 0 6.985 2.926c2.279.035 4.662-.856 6.502-2.256Zm-2.24-.971c-1.353.605-3.834 1.707-5.3 1.474a7.615 7.615 0 0 1-3.928-1.88l-.233-.22-8.873-8.788 4.41-4.373 13.925 13.787Z"
            />
          </svg>
          <input
            class="input"
            placeholder="Temps de préparation"
            type="text"
            value=""
          />
          <svg
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.8 7.64a.93.93 0 0 1 1.34 0c1.44 1.46 1.06 3.86.04 4.88-.6.6-.36 1.82 0 2.3.36.48.28 1.1 0 1.38-.18.18-.42.28-.66.28-.24 0-.48-.08-.66-.28-1.18-1.18-1.06-4 0-5.08.48-.48.5-1.52-.06-2.1-.36-.38-.36-1 0-1.38Zm6.52-2.6a.93.93 0 0 1 1.34 0c1.72 1.78 1.74 4.44.06 6.18-.98 1-1 1.96-.06 2.94.36.38.36 1 0 1.38-.18.18-.42.28-.66.28-.24 0-.48-.08-.66-.28-1.68-1.74-1.66-3.94.04-5.72.94-.96.9-2.4-.06-3.4-.36-.38-.36-1 0-1.38Zm7.76 2.62c1.44 1.44 1.06 3.86.06 4.86-.6.6-.36 1.8-.02 2.3.36.48.28 1.1 0 1.38-.18.18-.42.28-.66.28-.24 0-.48-.08-.66-.28-1.18-1.18-1.06-4 0-5.06.48-.48.5-1.52-.06-2.1-.36-.38-.36-1 0-1.38a.93.93 0 0 1 1.34 0Zm9.74 13.68h-1.08c.3-.7.56-1.42.76-2.16h-.02c0-.02.005-.04.01-.06.005-.02.01-.04.01-.06 0-.54-.44-.98-.98-.98H4.48c-.54 0-.98.44-.98.98 0 .04 0 .08.02.12H3.5c.2.74.46 1.46.76 2.16H3.18c-.54 0-.98.44-.98.98s.44.98.98.98h1.94c.04 0 .08 0 .12-.02 2.98 5.08 8.48 8.48 14.76 8.48 6.28 0 11.78-3.4 14.76-8.48.02 0 .04.005.06.01.02.005.04.01.06.01h1.94c.54 0 .98-.44.98-.98 0-.56-.44-.98-.98-.98ZM20 29.8c-6.46 0-11.98-4.06-14.18-9.76h28.36c-2.2 5.7-7.72 9.76-14.18 9.76Zm7.76 3.24H12.88c-.54 0-.98.44-.98.98s.44.98.98.98h14.88c.54 0 .98-.44.98-.98s-.44-.98-.98-.98Z"
            />
          </svg>
          <input
            class="input"
            placeholder="Temps de cuisson"
            type="text"
            value=""
          />
          <svg
            height="42.5"
            viewBox="0 0 32 34"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.753.853a.91.91 0 0 0-.955.07l.955-.07Zm-.23.444a.41.41 0 0 0-.43.03l-.002.002c-2.898 2.073-5.319 5.575-6.162 8.903-.587 2.327-.667 4.952-.678 6.702l-.002.307-.3-.064c-1.479-.316-2.359-1.573-2.863-2.689a9.3 9.3 0 0 1-.673-2.211.42.42 0 0 0-.216-.31.414.414 0 0 0-.377-.008h-.002c-.04.02-.878.438-1.73 1.681-.851 1.243-1.727 3.324-1.823 6.687v.008c-.014.218-.015.438-.015.665 0 6.479 5.272 11.75 11.75 11.75S27.75 27.479 27.75 21c0-.321-.023-.622-.045-.932l-.023-.313c-.36-4.693-2.541-7.635-4.5-10.272l-.02-.026c-1.8-2.425-3.412-4.594-3.412-7.79 0-.155-.088-.298-.227-.37Zm.23-.444a.918.918 0 0 1 .497.815c0 3.019 1.512 5.066 3.333 7.517l.007.009c1.96 2.638 4.218 5.678 4.59 10.523l.023.314c.023.308.047.627.047.97 0 6.754-5.496 12.249-12.25 12.249S3.75 27.755 3.75 21c0-.225.001-.457.015-.691.1-3.445 1-5.621 1.91-6.951.908-1.325 1.818-1.796 1.924-1.848a.914.914 0 0 1 .83.014l.002.001a.92.92 0 0 1 .477.679v.003l.002.01.006.035.024.13a8.844 8.844 0 0 0 .602 1.9c.444.985 1.143 1.964 2.212 2.324.02-1.749.123-4.25.69-6.497.874-3.447 3.366-7.047 6.356-9.187l.145.203M17.01 19.853l.13.445c.27.924.796 1.677 1.339 2.448l.004.006c.773 1.099 1.604 2.28 1.75 4.282m0 0v.004c.008.112.017.235.017.368 0 2.472-1.892 4.51-4.25 4.51s-4.25-2.038-4.25-4.51v-.022c0-.072 0-.152.005-.249.028-1.017.222-1.726.444-2.21l.226-.492.228.49c.367.788.991 1.45 1.975 1.45.23 0 .416-.186.416-.416 0-.949.02-2.07.263-3.091.22-.92.744-1.887 1.402-2.66l.3-.353m-.198 1.03c-.48.65-.85 1.397-1.018 2.1-.228.956-.249 2.023-.249 2.974 0 .506-.41.916-.916.916-1.021 0-1.714-.567-2.156-1.246-.113.385-.2.884-.217 1.526v.008c-.005.082-.005.15-.005.224v.02c0 2.227 1.697 4.012 3.75 4.012s3.75-1.785 3.75-4.011c0-.113-.008-.22-.016-.335-.135-1.86-.896-2.945-1.664-4.038l-.01-.012c-.452-.644-.934-1.328-1.249-2.139Z"
            />
            <path
              d="m27.954 20.05-.023-.314c-.366-4.772-2.588-7.763-4.549-10.402C21.567 6.891 20 4.781 20 1.668a.668.668 0 0 0-1.055-.543c-2.944 2.107-5.4 5.658-6.258 9.046-.596 2.358-.675 5.01-.686 6.761-2.718-.58-3.334-4.647-3.34-4.692a.664.664 0 0 0-.951-.506c-.143.07-3.5 1.775-3.695 8.586C4 20.547 4 20.773 4 21c0 6.617 5.384 12 12 12s12-5.383 12-12c0-.332-.023-.642-.046-.95ZM16 31.666c-2.206 0-4-1.912-4-4.261 0-.08 0-.161.005-.26.027-.99.215-1.667.422-2.117.386.83 1.078 1.594 2.2 1.594a.666.666 0 0 0 .667-.666c0-.95.02-2.045.256-3.033.21-.877.713-1.808 1.35-2.556.283.97.835 1.755 1.374 2.522.772 1.096 1.569 2.23 1.709 4.162.008.115.017.23.017.354 0 2.35-1.794 4.26-4 4.26Zm4.213-.871a5.712 5.712 0 0 0 1.12-3.39c0-.157-.01-.303-.03-.576-.157-2.176-1.103-3.521-1.939-4.708-.711-1.01-1.325-1.883-1.325-3.122a.667.667 0 0 0-1.068-.531c-1.279.966-2.345 2.592-2.718 4.145-.19.796-.256 1.657-.28 2.366-.462-.498-.605-1.42-.606-1.432a.668.668 0 0 0-.965-.503c-.168.086-1.644.932-1.728 4.045a5.713 5.713 0 0 0 1.114 3.706C7.995 29.159 5.333 25.386 5.333 21c0-.2 0-.398.013-.62.114-3.997 1.393-5.978 2.228-6.852.582 2.083 2.106 4.806 5.092 4.806a.666.666 0 0 0 .667-.667c0-2.23.05-4.809.646-7.17.677-2.674 2.532-5.562 4.77-7.517.372 2.856 1.928 4.95 3.562 7.15 1.944 2.615 3.953 5.32 4.29 9.7l.022.319c.021.277.043.554.043.851 0 4.386-2.663 8.159-6.454 9.796Z"
            />
          </svg>
          <input
            class="input"
            placeholder="Temps de cuisson"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="submitButton"
      >
        Enregistrer
      </button>
    </form>
  </div>
</div>
`;
